pkg/network/types.go

```diff
@@ -4,31 +4,41 @@ import (
 	"fmt"
 
 	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/autoscaling"
+	"github.com/aws/aws-sdk-go/service/autoscaling/autoscalingiface"
 	"github.com/aws/aws-sdk-go/service/ec2"
 	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"strings"
 )
 
 type Network struct {
 	InstanceIDs      []string
 	SecurityGroupIDs []string
 	SubnetIDs        []string
+	ASGNames         []string
 	Vpc              *ec2.Vpc
 }
 
-func GetNetworkInfoForEC2Instances(ec2svc ec2iface.EC2API, nodeInstanceIds []string) (vpcIds []string, subnetIds []string, securityGroups []string, err error) {
+func GetNetworkInfoForEC2Instances(ec2svc ec2iface.EC2API, autoscalingSvc autoscalingiface.AutoScalingAPI, nodeInstanceIds []string) (vpcIds []string, subnetIds []string, securityGroups []string, asgNames []string, err error) {
 	output, err := ec2svc.DescribeInstances(&ec2.DescribeInstancesInput{
 		InstanceIds: aws.StringSlice(nodeInstanceIds),
 	})
 	if err != nil {
-		return nil, nil, nil, fmt.Errorf("Error describing instances: %s", err)
+		return nil, nil, nil, nil, fmt.Errorf("Error describing instances: %s", err)
 	}
 
 	vids := map[string]bool{}
 	sids := map[string]bool{}
 	sgs := map[string]bool{}
+	asgs := map[string]bool{}
 
 	for _, reservation := range output.Reservations {
 		for _, instance := range reservation.Instances {
+			for _, tag := range instance.Tags {
+				if *tag.Key == "aws:autoscaling:groupName" {
+					asgs[*tag.Value] = true
+				}
+			}
 			if *instance.SubnetId != "" {
```

Comment:
Can this be removed ?

Reply:
It depends if there is a way to spin up worker nodes in EKS cluster outside of ASG.

Reply:
then user of this controller has to rely on subnetIDs for instance itself.

Reply:
Agreed, thats unlikely but a valid scenario.